board: sensry: ganymed_bob_sy120 - doc update (#87415)
zephyrproject-rtos/zephyr: boards/sensry/ganymed_bob/doc/index.rst

.. zephyr:board:: ganymed_bob

.. _ganymed_bob:

.. figure:: img/ganymed_bob_sy120_gbm.webp
     :align: center
     :alt: Ganymed Break-Out-Board (BOB) equipped with SY120 GBM

     Ganymed Break-Out-Board (BOB) equipped with SY120 GBM (Credit: Sensry)

.. figure:: img/ganymed_bob_sy120_gen1.webp
     :align: center
     :alt: Ganymed Break-Out-Board (BOB) equipped with SY120 GEN1

>      Ganymed Break-Out-Board (BOB) equipped with SY120 GEN1 (Credit: Sensry)

Reviewer: not looking really good, nor adding a lot of value IMO, could you just remove them?
see https://builds.zephyrproject.io/zephyr/pr/87415/docs/boards/sensry/ganymed_bob/doc/index.html

Author: ah thats cool. I can see a rendered preview. didnt know about that. thanks and agree, it doesnt look nice. will definitely remove it.

Author: done

* 512 KB Global MRAM
* 512 KB Secure MRAM
* CLOCK
* Peripherals

    * :abbr:`32x GPIO (General Purpose Input Output)`
    * :abbr:`4x TWIM (I2C-compatible two-wire interface with MicroDMA)`
    * 4x I2S
    * :abbr:`7x SPI (Serial Peripheral Interface with MicroDMA)`
    * :abbr:`3x UART (Universal receiver-transmitter with MicroDMA)`
    * :abbr:`1x TSN (Time sensitive networking ethernet MAC with MicroDMA)`
    * 1x CAN-FD
    * 3x ADC
* Power section for on-board power generation and power measurement (selectable)

    * USB type-C
    * external 5V power source
* 40-pin JTAG connector (compatible to Olimex ARM-JTAG-OCD-H)
* USB over FTDI (connected to UART0)
* Header for all I/Os and configuration

* Assembly options for the SoC

  * SY120-GBM - Generic Base Module without top level sensors
  * SY120-GEN1 - Generic Module type 1 with top level sensors (see below)

> 
> The ``ganymed-bob/sy120-gen1`` comes with additional hardware:
> 
> +-----------+------------+----------------------------+
> | Interface | Controller | Driver/Component           |
> +===========+============+============================+
> | BME680    | on-chip    | environment sensor (SPI1)  |
> +-----------+------------+----------------------------+
> | BMA456    | on-chip    | acceleration sensor (SPI0) |
> +-----------+------------+----------------------------+
> | BMG250    | on-chip    | gyrosope sensor (SPI2)     |
> +-----------+------------+----------------------------+
> | MIS2DH    | on-chip    | vibration sensor (I2C0)    |
> +-----------+------------+----------------------------+

Reviewer: I would just let the .. zephyr:board-supported-hw:: directive take care of this for you.

```suggestion

The ``ganymed-bob/sy120-gen1`` comes with additional on-board sensors.
```

* USB over FTDI (connected to UART0)
* Header for all I/Os and configuration

* Assembly options for the SoC

  * SY120-GBM - Generic Base Module without top level sensors
  * SY120-GEN1 - Generic Module type 1 with top level sensors (see below)

The ``ganymed-bob/sy120-gen1`` comes with additional on-board sensors.

Supported Features
==================

.. zephyr:board-supported-hw::

For more detailed description please refer to `Ganymed BreakOut Board Documentation`_

Programming and Testing
***********************

Building
========

Applications for the ``ganymed_bob/sy120_gbm`` board can be
built and flashed in the usual way. See
:ref:`build_an_application` and :ref:`application_run` for more details on
building and running.

Building the :zephyr:code-sample:`hello_world` sample:

.. zephyr-app-commands::
   :tool: west
   :zephyr-app: samples/hello_world
   :board: ganymed_bob/sy120_gbm
   :goals: build
   :compact:

Flashing
========

Test the Ganymed with a :zephyr:code-sample:`hello_world` sample.

Flash the zephyr image:

.. zephyr-app-commands::
   :tool: west
   :zephyr-app: samples/hello_world
   :goals: flash
   :west-args: --serial /dev/ttyUSB0

>    :compact:

Reviewer: this is actually incorrect - have you tested it? The board is missing a board.cmake file so does in fact not support west flash / debug. Can you please address in a follow-up PR? @tswaehn

Author: yes, correct, should `--dev-id `

Author: please note that with the PR  https://github.com/zephyrproject-rtos/zephyr/pull/87394, the cmake file for this board will be added.

Reviewer: I would just fix this in the other PR. There's no point in doing another PR/commit for something that will still be broken until the board.cmake gets added

* 512 KB Global MRAM
* 512 KB Secure MRAM
* CLOCK

> * peripherals
>     * :abbr:`32x GPIO (General Purpose Input Output)`

Reviewer: ```suggestion
* Peripherals

    * :abbr:`32x GPIO (General Purpose Input Output)`
```

    * :abbr:`4x TWIM (I2C-compatible two-wire interface with MicroDMA)`
    * 4x I2S
    * :abbr:`7x SPI (Serial Peripheral Interface with MicroDMA)`
    * :abbr:`3x UART (Universal receiver-transmitter with MicroDMA)`
    * :abbr:`1x TSN (Time sensitive networking ethernet MAC with MicroDMA)`
    * 1x CAN-FD
    * 3x ADC

> * power section for on-board power generation and power measurement (selectable)

Reviewer: ```suggestion
* Power section for on-board power generation and power measurement (selectable)

```

+-----------+------------+----------------------+
| UART      | on-chip    | serial               |
+-----------+------------+----------------------+
| TSN       | on-chip    | ethernet MAC         |
+-----------+------------+----------------------+
| MDIO      | on-chip    |                      |
+-----------+------------+----------------------+
| TIMER     | on-chip    |                      |
+-----------+------------+----------------------+
| PINCTRL   | on-chip    |                      |
+-----------+------------+----------------------+
| I2S       | on-chip    | coming soon          |
+-----------+------------+----------------------+
| CAN       | on-chip    | CAN - coming soon    |
+-----------+------------+----------------------+
| SPU       | on-chip    | system protection    |
+-----------+------------+----------------------+
| GRTC      | on-chip    | counter              |
+-----------+------------+----------------------+
| PWM       | on-chip    | pwm                  |
+-----------+------------+----------------------+
| MRAM      | on-chip    | non-volatile memory  |
+-----------+------------+----------------------+
| SAADC     | on-chip    | adc - coming soon    |

> +-----------+------------+----------------------+

Reviewer 2: use ``.. zephyr:board-supported-hw::``

and the ``.. zephyr:board::`` at the top

Author: replaced with your suggestions

Programming and Debugging
*************************

Applications for the ``ganymed_bob/sy120_gbm`` board can be
built and flashed in the usual way. See
:ref:`build_an_application` and :ref:`application_run` for more details on
building and running.

Building the :zephyr:code-sample:`hello_world` sample:

> .. code-block:: console

Reviewer 3: replace with
```
.. zephyr-app-commands::
   :tool: west
   :zephyr-app: samples/hello_world
   :board: ganymed_bob/sy120_gbm
   :goals: build
   :compact:
```

Author: replaced with your suggestions

    west build -b ganymed_bob/sy120_gbm samples/hello_world

Testing the Ganymed BreakOut Board
**********************************

Test the Ganymed with a :zephyr:code-sample:`hello_world` sample.

Flash the zephyr image:

.. code-block:: console

>     west flash --serial /dev/ttyUSB0

Reviewer 3: should also use above syntax

Author: replaced with your suggestions
